File: JS/遍历/for.js
/*
 * @Author: your name
 * @Date: 2021-03-07 22:48:34
 * @LastEditTime: 2021-03-13 00:49:22
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: /ivan个人练习代码/JS/遍历/for.js
 */

// function CheckIn(){
//     this.name = 'cjz'
// }
// CheckIn.prototype.job = 'enineer'
// const obj = new CheckIn()
// console.log(obj)
// for(let key in obj){
//     console.log(key)
// }

const arr = ['a',123,{a:'1',b:'2'}]
arr.name = 'demo' // in还会遍历自定义属性
for(let key in arr){
    console.log(key)//只是索引，推荐是是用来遍历对象的属性
}
for(let value of arr){
    console.log(value) //这个直接是拿来遍历
}

class RangeIterator {
    constructor(start, stop) {
      this.value = start;
      this.stop = stop;
    }
  
    [Symbol.iterator]() { return this; }
  
    next() {
      var value = this.value;
      if (value < this.stop) {
        this.value++;
        return {done: false, value: value};
      }
      return {done: true, value: undefined};
    }
  }
  
  function range(start, stop) {
    return new RangeIterator(start, stop);
  }
  
const makeLit = new range(0,3)
console.log(makeLit[Symbol.iterator]().next())
console.log(makeLit[Symbol.iterator]().next())
console.log(makeLit[Symbol.iterator]().next())

//   for (var value of range(0, 3)) {
//     console.log(value); // 0, 1, 2
//   }

const arrReview = [1,3,3,4,5,1]
const Obj = {
  [Symbol.iterator]() { return this; },
  name: 'cjz',
  value: 'tencent',
  job: 'fontEnd',
  time: 1,
  next() {
    if(this.time < 10){
      this.time++
     return { value: 1, done: false }
    }else{
       return { done: true, value: undefined}
    }
}
}
for(let i of arrReview){
  console.log(i)
}
for(let value of Obj){
  console.log(value,'chh')
}

File: 编译原理/for.cpp
/*
 * @Author: your name
 * @Date: 2021-03-12 10:30:14
 * @LastEditTime: 2021-04-18 16:18:13
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: /ivan个人练习代码/编译原理/for.cpp
 */
#include <stdio.h>
#include <string>
#include <stdlib.h>
#include <iostream>
#include <stack>
using namespace std;
 
int opration(int a, char op, int b);
int compare(char str[]);
 
int main()
{
	char str[100];
	while (1)
	{
		gets(str);
		printf("%d\n", compare(str));
	}
		
	return 0;
}
 
int opration(int a, char op, int b)
{
	switch (op)
	{
	case '+':return a + b; break;
	case '-':return a - b; break;
	case '*':return a*b; break;
	case '/':return a / b; break;
	}
}
int compare(char str[])
{
	stack<int> s;//暂存操作数
	char op;//暂存操作符
	int a, b;
	for (int i = 0; i < strlen(str); i++)
	{
		if (str[i] >= '0'&&str[i] <= '9')
			s.push(str[i] - '0');
		else
		{
			op = str[i];
			b = s.top();
			s.pop();
			a = s.top();
			s.pop();
			s.push(opration(a,op,b));//将当前计算结果入栈
		}
	}
	return s.top();
}